lib/src/merge.rs

```diff
@@ -816,6 +882,100 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_deduplicate() {
+        // 1-way merge
+        assert_eq!(c(&[], &[0]).deduplicate(), c(&[], &[0]));
+        // 3-way merge
+        assert_eq!(c(&[0], &[0, 0]).deduplicate(), c(&[0], &[0, 0]));
+        assert_eq!(c(&[0], &[0, 1]).deduplicate(), c(&[0], &[0, 1]));
+        assert_eq!(c(&[0], &[1, 0]).deduplicate(), c(&[0], &[1, 0]));
+        assert_eq!(c(&[0], &[1, 1]).deduplicate(), c(&[0], &[1, 1]));
+        assert_eq!(c(&[0], &[1, 2]).deduplicate(), c(&[0], &[1, 2]));
+        // 5-way merge
+        assert_eq!(c(&[0, 0], &[0, 0, 0]).deduplicate(), c(&[0], &[0, 0]));
+        assert_eq!(c(&[0, 0], &[0, 0, 1]).deduplicate(), c(&[0], &[0, 1]));
+        assert_eq!(c(&[0, 0], &[0, 1, 0]).deduplicate(), c(&[0], &[0, 1]));
+        assert_eq!(c(&[0, 0], &[0, 1, 1]).deduplicate(), c(&[0], &[0, 1]));
+        assert_eq!(c(&[0, 0], &[0, 1, 2]).deduplicate(), c(&[0, 0], &[0, 1, 2]));
+        assert_eq!(c(&[0, 0], &[1, 0, 0]).deduplicate(), c(&[0], &[1, 0]));
+        assert_eq!(c(&[0, 0], &[1, 0, 1]).deduplicate(), c(&[0], &[1, 0]));
+        assert_eq!(c(&[0, 0], &[1, 0, 2]).deduplicate(), c(&[0, 0], &[1, 0, 2]));
+        assert_eq!(c(&[0, 0], &[1, 1, 0]).deduplicate(), c(&[0], &[1, 0]));
+        assert_eq!(c(&[0, 0], &[1, 1, 1]).deduplicate(), c(&[0], &[1, 1]));
+        assert_eq!(c(&[0, 0], &[1, 1, 2]).deduplicate(), c(&[0], &[1, 2]));
+        assert_eq!(c(&[0, 0], &[1, 2, 0]).deduplicate(), c(&[0, 0], &[1, 2, 0]));
+        assert_eq!(c(&[0, 0], &[1, 2, 1]).deduplicate(), c(&[0], &[1, 2]));
```

Comment:
I don't know whether this should be deduplicated, for the same reason as my other long comment.

In my mind, all that's needed for per-file simplification is simplifying "-Y + Y" to nothing. This might not even require a "deduplicate" function, regular simplification might be sufficient. I might also be missing something, I'm unsure.